specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -252,867 +202,83 @@ export function WbViewReact({
+            onDatasetRefresh={triggerDatasetRefresh}
+            mappings={mappings!}
+            checkDeletedFail={checkDeletedFail}
+            onSpreadsheetUpToDate={spreadsheetUpToDate}
+            workbench={workbench}
+            onToggleResults={toggleResults}
           />
         </div>
-        <Button.Small
-          aria-haspopup="grid"
-          aria-pressed={showToolkit}
-          onClick={toggleToolkit}
-        >
-          {commonText.tools()}
-        </Button.Small>
-        <span className="-ml-1 flex-1" />
-        {canUpdate || isMapped ? (
-          <Link.Small href={`/specify/workbench/plan/${dataset.id}/`}>
-            {wbPlanText.dataMapper()}
-          </Link.Small>
-        ) : undefined}
-        <WbActionsComponent
-          dataset={dataset}
-          hasUnsavedChanges={hasUnsavedChanges}
-          isUploaded={isUploaded}
-          onRefresh={triggerRefresh}
-          mappings={mappings!}
-          checkDeletedFail={checkDeletedFail}
-          onSpreadsheetUpToDate={spreadsheetUpToDate}
-          workbench={workbench}
-          onToggleResults={toggleResults}
-        />
-      </div>
-      {showToolkit ? (
-        <WbToolkit
-          dataset={dataset}
-          hotRef={hotRef}
-          mappings={mappings!}
-          data={data}
-          handleDatasetDelete={handleDatasetDelete}
-          hasUnsavedChanges={hasUnsavedChanges}
-          triggerRefresh={triggerRefresh}
-        />
-      ) : undefined}
-      <div className="flex flex-1 gap-4 overflow-hidden">
-        <section className="flex-1 overflow-hidden overscroll-none">
-          <WbSpreadsheet
+        {showToolkit ? (
+          <WbToolkit
             dataset={dataset}
-            hotRef={hotRef}
-            isUploaded={isUploaded}
-            data={data}
-            validation={workbench.validation}
-            cells={workbench.cells}
-            disambiguation={workbench.disambiguation}
-            hooks={hooks}
+            hot={hot}
             mappings={mappings!}
+            data={data}
+            onDatasetDeleted={handleDatasetDeleted}
+            hasUnsavedChanges={hasUnsavedChanges}
+            triggerDatasetRefresh={triggerDatasetRefresh}
           />
-        </section>
-        {showResults ? (
-          <aside aria-live="polite">
-            <WbUploaded
-              dataSetId={dataset.id}
-              dataSetName={dataset.name}
+        ) : undefined}
+        <div className="flex flex-1 gap-4 overflow-hidden">
+          <section className="flex-1 overflow-hidden overscroll-none">
+            <WbSpreadsheet
+              dataset={dataset}
+              setHotTable={setHotTable as React.Ref<HotTable>}
+              hot={hot}
               isUploaded={isUploaded}
-              recordCounts={workbench.validation.uploadResults.recordCounts}
-              onClose={closeResults}
+              data={data}
+              workbench={workbench}
+              mappings={mappings!}
```

Comment:
`!` here is possibly a mistake - WbSpreadsheet can be rendered with undefined mappings and should be able to handle that (render as much content as possible even with mappings undefined, as we are supporting editing datasets before they are mapped)